thirdparty/CMakeLists.txt

```diff
@@ -18,6 +18,23 @@ endif()
 set(CNMEM_INCLUDE_DIR "${cnmem_SOURCE_DIR}/include" PARENT_SCOPE)
 set(CNMEM_SOURCE_DIR "${cnmem_SOURCE_DIR}/src" PARENT_SCOPE)
 
+###################################################################################################
+# - cub -------------------------------------------------------------------------------------------
+
+FetchContent_Declare(
+    cub
+    GIT_REPOSITORY https://github.com/thrust/cub
+    GIT_TAG        1.9.10
+    GIT_SHALLOW    true
+)
+
+FetchContent_GetProperties(cub)
+if(NOT cub_POPULATED)
+  FetchContent_Populate(cub)
+  # We are not using the cub CMake targets, so no need to call `add_subdirectory()`.
```

Comment:
Is this comment necessary?

Reply:
In the doc for `FetchContent` it shows a common pattern of populating and then adding subdirectory. I guess this is more of a reminder for ourselves why we are not doing that.